enumerable.rb

```diff
@@ -0,0 +1,85 @@
+module Enumerable
+    def my_each
+        for item in self
+            yield(item)
+        end
+        self
+    end
+   
+    def my_each_with_index
+        i = 0
+        while i < self.length
+         yield(self[i], i)
+        i += 1
+    end
+    self
+    end
+
+    def my_select
+        new_array = []
+        self.my_each do |item|
+          new_array.push(item) if yield(item)
+        end
+        new_array
+    end
+
+    def my_all?
+        test = true
+        self.my_each do |item|
+          test = false if !yield(item)
+          break
+        end
+        return test
+    end
+    
+    def my_any?
+        test = false
+        self.my_each do |item|
+          test = true if yield(item)
+          break
+        end
+        return test
+    end
+    
+    def my_none?
+        test = true
+        self.my_each do |item|
+          test = false if yield(item)
+          break
+        end
+        return test
+    end
+    
+    def my_count
+        total = 0
+        if block_given?
+          self.my_each {|item| total += 1 if yield(item)}
+        else
+          total = self.count
+        end
+        return total
+    end
```

Comment:
- [ ] returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
 - [ ] counts the number of items in enum that are equal to item if an argument is given